hive3/src/main/java/org/apache/iceberg/mr/hive/vector/CompatibilityHiveVectorUtils.java

```diff
@@ -67,12 +66,13 @@ public static MapWork findMapWork(JobConf job) {
       LOG.debug("Initializing for input {}", inputName);
     }
     String prefixes = job.get(DagUtils.TEZ_MERGE_WORK_FILE_PREFIXES);
-    if (prefixes != null && !StringUtils.isBlank(prefixes)) {
-      // Currently SMB is broken, so we cannot check if it's  compatible with IO elevator.
-      // So, we don't use the below code that would get the correct MapWork. See HIVE-16985.
-      return null;
+    if (prefixes != null) {
+      if (!prefixes.trim().isEmpty()) {
```

Comment:
both ifs can be folded into a single if like in the previous version of the code

Reply:
agreed, I have tested all the combinations just to be sure in a JShell 👍 . 

`import org.apache.commons.lang3.StringUtils;
String a=null;

if (a != null && !StringUtils.isBlank(a)) {
    System.out.println("nullllll check");
    }
if (a != null && !a.trim().isEmpty()) {
        System.out.println("null check native");
    }
a="";
if (a != null && !StringUtils.isBlank(a)) {
        System.out.println("empty check ");
    }
if (a != null && !a.trim().isEmpty()) {
        System.out.println("empty check native");
    }
a=" ";
if (a != null && !StringUtils.isBlank(a)) {
        System.out.println("blank check");
    }
if (a != null && !a.trim().isEmpty()) {
        System.out.println("blank check native");
    }
a="bob";
if (a != null && !StringUtils.isBlank(a)) {
        System.out.println("things check");
    }
    if (a != null && !a.trim().isEmpty()) {
        System.out.println("things check native");
    }
a="  bob  ";
if (a != null && !StringUtils.isBlank(a)) {
        System.out.println("trim things check");
    }
    if (a != null && !a.trim().isEmpty()) {
        System.out.println("trim things check native");
    }`